src/databricks/labs/ucx/hive_metastore/mapping.py

```diff
@@ -191,20 +191,41 @@ def _get_table_in_scope_task(self, table_to_migrate: TableToMigrate) -> TableToM
 
         return table_to_migrate
 
+    def _get_table_properties(self, table):
+        if table.is_table_in_mount:
+            return self._sql_backend.fetch(f"SHOW TBLPROPERTIES delta.`{table.location}`")
+        return self._sql_backend.fetch(
+            f"SHOW TBLPROPERTIES {escape_sql_identifier(table.database)}.{escape_sql_identifier(table.name)}"
+        )
+
     def exists_in_uc(self, src_table: Table, target_key: str):
         # Attempts to get the target table info from UC returns True if it exists.
-        try:
-            table_info = self._ws.tables.get(target_key)
-            if not table_info.properties:
-                return True
-            upgraded_from = table_info.properties.get("upgraded_from")
-            if upgraded_from and upgraded_from != src_table.key:
-                raise ResourceConflict(
-                    f"Expected to be migrated from {src_table.key}, but got {upgraded_from}. "
-                    "You can skip this error using the CLI command: "
-                    "databricks labs ucx skip "
-                    f"--schema {src_table.database} --table {src_table.name}"
-                )
+        table_info = self._try_get_table_in_uc(target_key)
```

Comment:
Here we just check if the target table is in UC or not. It's possible an UC table with different name took this location already and fail the table creation later. But there is no easy way to check if a location is already occupied in UC, so better to try catch the the table creation DDL execution.

Reply:
Indeed, this currently happens at my customer, the good thing is that the code throws a PERMISSION_DENIED error with the table name in UC that cause the issue, so it can be easily handled.

Reply:
Additionally we're having a situation where tables Delta tables are corrupted, and thus we aren't able to create tables there